base/linalg/matmul.jl

```diff
@@ -44,7 +44,7 @@ end
 vecdot{T<:BlasReal}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dot(x, y)
 vecdot{T<:BlasComplex}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dotc(x, y)
 
-function dot{T<:BlasReal, TI<:Integer}(x::Vector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::Vector{T}, ry::Union{UnitRange{TI},Range{TI}})
+function dot{T<:BlasReal, TI<:Integer}(x::StridedVector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::StridedVector{T}, ry::Union{UnitRange{TI},Range{TI}})
```

Comment:
@andreasnoack Okay, so I thought about it. Maybe I should just revert the definition, and only run the `dot(vector, range, vector, range)` method for `Array`s, and not `SubArray`s. This will mean that `dot(a::SubArray, b::Range, c::SubArray, d::Range)` won't have a method defined, but maybe that actually makes sense as you hinted at above. Why not just create a different sub with the appropriate shape if that is what you want. I'll make this change, and you can comment if you agree or not.